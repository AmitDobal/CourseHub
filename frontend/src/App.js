import { BrowserRouter, Route, Routes } from "react-router-dom";
import "./App.css";
import Home from "./pages/Home";
import Login from "./pages/Login";
import Register from "./pages/Register";
import Navbar from "./components/Navbar";
import Error from "./pages/Error";
import Courses from "./pages/Courses";
import Footer from "./components/Footer";
import { Box } from "@mui/material";
import Dashboard from "./pages/Dashboard";

function App() {
  return (
    <BrowserRouter>
      <Navbar />
      <Box
        minHeight="90vh" // Make the container at least the height of the viewport
        display="flex"
        flexDirection="column">
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/login" element={<Login />} />
          <Route path="/register" element={<Register />} />
          <Route path="/courses" element={<Courses />} />
          <Route path="/dashboard" element={<Dashboard  />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </Box>
      <Footer />
    </BrowserRouter>
  );
}

export default App;
