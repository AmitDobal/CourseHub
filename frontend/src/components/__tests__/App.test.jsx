import { render, screen } from '@testing-library/react';
import App from '../App';

describe('App component', () => {
it("should render app with error", () =>{
  const {asFragment} = render(<App/>);
  expect(asFragment()).toMatchSnapshot();
})
});

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/Hello, friend/i);
//   expect(linkElement).toBeInTheDocument();
// });
