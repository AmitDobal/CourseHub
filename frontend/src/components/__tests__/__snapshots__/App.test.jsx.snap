// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app without error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16ci0ji"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
              >
                Course Store
              </h6>
            </div>
          </header>
        </div>
        <div
          class="MuiBox-root css-1afphko"
        >
          <div
            class="MuiBox-root css-ksvmh9"
          >
            <div
              class="MuiBox-root css-1carym4"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-thsn50-MuiPaper-root"
              >
                Course filter
              </div>
            </div>
            <div
              class="MuiBox-root css-1x97c6v"
            >
              <div
                class="MuiBox-root css-19t1w01"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
                  data-testid="course-loader"
                  style="width: 80%; height: 200px;"
                />
              </div>
              <div
                class="MuiBox-root css-1mf0mzh"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <footer
      class="MuiBox-root css-146ovje"
    >
      <div
        class="MuiBox-root css-vdfig3"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Cours0
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Â©2023 Cours0, Inc.
          </p>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
